dcos/http.py

```diff
@@ -180,3 +220,29 @@ def delete(url, **kwargs):
 def silence_requests_warnings():
     """Silence warnings from requests.packages.urllib3.  See DCOS-1007."""
     requests.packages.urllib3.disable_warnings()
+
+
+def _get_http_auth_credentials(response):
+    """Get username/password for HTTP Basic Auth
+
+    :param response: requests.response
+    :type response: requests.Response
+    :returns: process return code
+    :rtype: int
+    """
+
+    parsed_url = urlparse(response.url)
+    hostname = parsed_url.hostname
+    if response.headers.get('www-authenticate') == 'Basic realm="Restricted"':
+        username = parsed_url.username
+        if username is None:
+            sys.stdout.write('Username: ')
+            sys.stdout.flush()
+            username = sys.stdin.readline().strip().lower()
+        password = getpass.getpass()
+        AUTH_HOSTS[hostname] = HTTPBasicAuth(username, password)
+    else:
+        raise DCOSException("We don't support this authentication. {}".format(
```

Comment:
`Server responded with an HTTP "www-authenticate" field of {}.  We only support {}."

and

`Invalid HTTP response: server returned an HTTP 401 response with no "www-authenticate" field`